src/vs/workbench/api/common/shared/tasks.ts

```diff
@@ -78,6 +78,7 @@ export interface ITaskSourceDTO {
 	scope?: number | UriComponents;
 	color?: string;
 	icon?: string;
+	hide?: boolean;
```

Comment:
This would be a good case for a smoke test; set the tasks.json without hide and make sure it exists in the quick pick, add hide and make sure it's hidden.

Reply:
Should be pretty easy to setup a new suite, copy smoke\src\areas\terminal\terminal.test.ts into tasks.test.ts and setup the suite there. This would only be interacting with tasks.json and the runTask command so I don't think there would be are problems. `app.workbench.quickinput.waitForQuickInputElements` can be used to verify an element is within or not.

Can do as a follow up, we just don't want to see it stick around beyond debt week which has happened with a few other test defer issues.

Reply:
https://github.com/microsoft/vscode/issues/154277